Library/Formula/postgresql.rb

```diff
@@ -77,8 +82,41 @@ def install
   end
 
   def post_install
-    unless File.exist? "#{var}/postgres"
-      system "#{bin}/initdb", "#{var}/postgres"
+    if File.exist? "#{var}/postgres/PG_VERSION"
+      pg_version = (var/"postgres/PG_VERSION").read.chomp
+    end
+
+    if (var/"postgres").exist? && (var/"postgres-#{pg_version}").exist?
+      puts "We have not attempted to backup your data due to your multi-postgres setup."
+    end
+
+    # With the 9.5 release, we'll need to adapt logic to make sure we preserve the
+    # 9.4 var directories, without breaking postgres. This should be able to be done
+    # by modifying some lines below to var/"postgres-9.4" instead of var/"postgres"
+    current_data = var/"postgres"
+    backup_path = var/"postgres-#{version_suffix}"
+    # This should always be the last full version shipped by Brew, i.e. 9.4.0.
+    prior_ver = "9.4.0"
```

Comment:
Use Dir[HOMEBREW_CELLAR/"#{pg_version}.*"].first to get the previous installed version (if there is one, there might not be for e.g.a fresh install)